options/options.go

```diff
@@ -52,10 +52,17 @@ type interpreter struct {
 	file                    file
 	resolver                linker.Resolver
 	overrideDescriptorProto linker.File
-	lenient                 bool
-	reporter                *reporter.Handler
-	index                   sourceinfo.OptionIndex
-	pathBuffer              []int32
+
+	index      sourceinfo.OptionIndex
+	pathBuffer []int32
+
+	reporter internal.ErrorHandler
+	lenient  bool
+
+	// lenienceEnabled is set to true when errors reported to reporter
+	// should be lenient
+	lenienceEnabled    bool
+	lenientErrReported bool
```

Comment:
I would have liked to have formatted this with less churn in the diff. But then I got `govet` lint warnings based on the struct size being inflated due to `bool` fields in the middle of the struct. So I moved them all to the end to get rid of that lint issue.